components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/handlers/APIEndpointPswdRegistryHandler.java

```diff
@@ -0,0 +1,126 @@
+import org.wso2.carbon.registry.core.jdbc.handlers.RequestContext;
+import org.wso2.carbon.registry.core.utils.RegistryUtils;
+
+import javax.xml.stream.XMLStreamException;
+import java.util.Iterator;
+
+public class APIEndpointPswdRegistryHandler extends Handler {
+
+    private static Log log = LogFactory.getLog(APIEndpointPswdRegistryHandler.class);
+
+    private static final String OVERVIEW_ELEMENT = "overview";
+    private static final String ENDPOINT_PSWD_ELEMENT = "endpointPpassword";
+    private static final String REGISTRY_HIDDEN_PROPERTY = "registry.HiddenEpProperty";
+    private static final String DEFAULT_PSWD = "*****"; //5 stars
+
+    /**
+     * This method is called for a registry put operation
+     *
+     * @param requestContext - The request context
+     * @throws RegistryException
+     */
+    public void put(RequestContext requestContext) throws RegistryException {
+        Resource resource = requestContext.getResource();
+        Object content = resource.getContent();
+        String resourceContent;
+        if (content instanceof String) {
+            resourceContent = (String) resource.getContent();
+        } else if (content instanceof byte[]) {
+            resourceContent = RegistryUtils.decodeBytes((byte[]) resource.getContent());
+        } else {
+            return;
+        }
+        try {
+            OMElement omElement = AXIOMUtil.stringToOM(resourceContent);
+            Iterator mainChildIt = omElement.getChildren();
+            while (mainChildIt.hasNext()) {
+                Object childObj = mainChildIt.next();
+                if ((childObj instanceof OMElement) && (OVERVIEW_ELEMENT
+                        .equals(((OMElement) childObj).getLocalName()))) {
+                    Iterator childIt = ((OMElement) childObj).getChildrenWithLocalName(ENDPOINT_PSWD_ELEMENT);
+                    //There is only one ep-password, hence no iteration
+                    if (childIt.hasNext()) {
+                        OMElement child = (OMElement) childIt.next();
+                        String pswd = child.getText();
+                        //Password has been edited on put
+                        if (!"".equals(pswd) && !DEFAULT_PSWD.equals(pswd)) {
+                            resource.setProperty(REGISTRY_HIDDEN_PROPERTY, pswd);
+                            child.setText(pswd);
+                        } else if (DEFAULT_PSWD.equals(pswd)) { //Password not being changed
+                            String actualPassword = resource.getProperty(REGISTRY_HIDDEN_PROPERTY);
+                            child.setText(actualPassword);
+                        }
+                    }
+                }
+            }
+            resource.setContent(RegistryUtils.encodeString(omElement.toString()));
+            requestContext.setResource(resource);
+        } catch (XMLStreamException x) {
+            log.error("There was an error in reading XML Stream");
```

Comment:
Here only logged the error. What is the reason for only logging it? why not throw the exception?